### PR TITLE
Channels style UUID

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR switches ID creation to use Channels style IDs (base32 encoded, 16 byte UUID).  Also, IDs are prefixed with a character denoting the id type for easy recognition in logs.

#### Ticket Link
https://community.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/47aa9bb4-6967-4a96-83c7-11bd6b20f1eb/150d1464-6d59-4c39-810f-ee201439dbaf?c=25e17a4b-34d9-4239-8cac-26638084835f

### CODE REVIEW

File: server/server/server.go [hunk 1]
```diff
@@ -170,8 +169,8 @@ func New(params Params) (*Server, error) {
 	// Init telemetry
 	telemetryID := settings["TelemetryID"]
 	if len(telemetryID) == 0 {
-		telemetryID = uuid.New().String()
-		if err = params.DBStore.SetSystemSetting("TelemetryID", uuid.New().String()); err != nil {
+		telemetryID = appModel.NewID("")
+		if err = params.DBStore.SetSystemSetting("TelemetryID", telemetryID); err != nil {
```
Comment (wiggin77) [inline server/server/server.go:173]: This is unrelated but looked like a bug...  the telemetry id that was generated and used was not the same id stored in system settings.

File: server/model/util.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	case "user":
+		prefix = "u"
+	default:
+		prefix = "8"
+	}
+	return prefix + mm_model.NewId()
+}
+
+// GetMillis is a convenience method to get milliseconds since epoch.
+func GetMillis() int64 {
```
Comment (wiggin77) [inline server/model/util.go:38]: These time methods are needed for next PR.  Was going to combine the millisecond timestamp PR with this one, but then saw the number of files changed and decided to keep them separate (except for the utils file).

File: webapp/package.json [hunk 3]
```diff
@@ -51,7 +51,8 @@
 		"react-redux": "^7.2.4",
 		"react-router-dom": "^5.2.0",
 		"react-select": "^4.3.0",
-		"react-simplemde-editor": "^5.0.1"
+		"react-simplemde-editor": "^5.0.1",
+		"to-data-view": "^2.0.0"
```
Comment (jespino) [inline webapp/package.json:55]: Is this change related to the PR?

File: webapp/package-lock.json [hunk 4]
```diff
@@ -1,18758 +1,8 @@
 {
 	"name": "focalboard",
 	"version": "0.9.1",
-	"lockfileVersion": 2,
+	"lockfileVersion": 1,
```
Comment (jespino) [inline webapp/package-lock.json:4]: Looks like you are accidentally rolling back to version 1 of npm (probably because you are not using the last version of node locally).

Comment (wiggin77) [reply webapp/package-lock.json:4]: Ugh.  Ok, I'll revert package-lock

File: webapp/src/mutator.ts [hunk 5]
```diff
@@ -250,7 +250,7 @@ class Mutator {
         }
         const srcTemplate = newBoard.fields.cardProperties[index]
         const newTemplate: IPropertyTemplate = {
-            id: Utils.createGuid(),
+            id: Utils.createGuid(''),
```
Comment (jespino) [inline webapp/src/mutator.ts:253]: I think this ends up being a block, shouldn't it have the `block` type?

File: webapp/src/mutator.ts [hunk 6]
```diff
@@ -436,7 +436,7 @@ class Mutator {
                         let option = newTemplate.options.find((o: IPropertyOption) => o.value === oldValue)
                         if (!option) {
                             option = {
-                                id: Utils.createGuid(),
+                                id: Utils.createGuid(''),
```
Comment (jespino) [inline webapp/src/mutator.ts:439]: same here, I think this is a `block`

File: webapp/src/mutator.ts [hunk 7]
```diff
@@ -205,7 +205,7 @@ class Mutator {
         }
 
         const newTemplate = template || {
-            id: Utils.createGuid(),
+            id: Utils.createGuid(''),
```
Comment (jespino) [inline webapp/src/mutator.ts:208]: same here, I think this ends up being a `block`.

File: webapp/src/components/table/table.test.tsx [hunk 8]
```diff
@@ -383,7 +383,7 @@ describe('components/table/Table extended', () => {
     test('should match snapshot with UpdatedBy', async () => {
         const board = TestBlockFactory.createBoard()
 
-        const modifiedById = Utils.createGuid()
+        const modifiedById = Utils.createGuid('')
```
Comment (jespino) [inline webapp/src/components/table/table.test.tsx:386]: I know this is tests, but it shouldn't be a `user`? and in the other cases in this file.

File: webapp/src/components/shareBoardComponent.tsx [hunk 9]
```diff
@@ -56,7 +56,7 @@ const ShareBoardComponent = React.memo((props: Props): JSX.Element => {
         const accept = window.confirm(intl.formatMessage({id: 'ShareBoard.confirmRegenerateToken', defaultMessage: 'This will invalidate previously shared links. Continue?'}))
         if (accept) {
             const newSharing: ISharing = sharing || createSharingInfo()
-            newSharing.token = Utils.createGuid()
+            newSharing.token = Utils.createGuid('')
```
Comment (jespino) [inline webapp/src/components/shareBoardComponent.tsx:59]: I guess this id should be a token id

File: webapp/src/components/shareBoardComponent.tsx [hunk 10]
```diff
@@ -38,7 +38,7 @@ const ShareBoardComponent = React.memo((props: Props): JSX.Element => {
         const newSharing: ISharing = {
             id: props.boardId,
             enabled: true,
-            token: Utils.createGuid(),
+            token: Utils.createGuid(''),
```
Comment (jespino) [inline webapp/src/components/shareBoardComponent.tsx:41]: same here

File: webapp/src/components/propertyValueElement.tsx [hunk 11]
```diff
@@ -110,7 +110,7 @@ const PropertyValueElement = (props:Props): JSX.Element => {
                 onCreate={
                     async (newValue, currentValues) => {
                         const option: IPropertyOption = {
-                            id: Utils.createGuid(),
+                            id: Utils.createGuid(''),
```
Comment (jespino) [inline webapp/src/components/propertyValueElement.tsx:113]: I think the ones in this files are `blocks`

**Review by jespino — CHANGES_REQUESTED** [review]
Missing some types of tokes on the frontend side, also other small comments around the javascript part (specially about package-lock.json and package.json)

File: webapp/package.json (hunk 3 above)
Comment (wiggin77) [reply webapp/package.json]: I reverted package.json.  It is no longer changed in the PR.

File: webapp/package-lock.json (hunk 4 above)
Comment (wiggin77) [reply webapp/package-lock.json]: Reverted via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/mutator.ts (hunk 5 above)
Comment (wiggin77) [reply webapp/src/mutator.ts]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/mutator.ts (hunk 6 above)
Comment (wiggin77) [reply webapp/src/mutator.ts]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/mutator.ts (hunk 7 above)
Comment (wiggin77) [reply webapp/src/mutator.ts]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/components/table/table.test.tsx (hunk 8 above)
Comment (wiggin77) [reply webapp/src/components/table/table.test.tsx]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/components/shareBoardComponent.tsx (hunk 9 above)
Comment (wiggin77) [reply webapp/src/components/shareBoardComponent.tsx]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/components/shareBoardComponent.tsx (hunk 10 above)
Comment (wiggin77) [reply webapp/src/components/shareBoardComponent.tsx]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/03daaf722a0bff201708cf27b0434c8fbfef4f4c

File: webapp/src/components/propertyValueElement.tsx (hunk 11 above)
Comment (wiggin77) [reply webapp/src/components/propertyValueElement.tsx]: Fixed via https://github.com/mattermost/focalboard/pull/1369/commits/155dab31629dc0b9728d6d730187a8f2d6563244

**Review by jespino — APPROVED** [review]
LGTM
